honeybee_energy/boundarycondition.py

```diff
@@ -1,96 +1,10 @@
-"""Energy Face Outside Boundary Condition."""
+"""Extra Boundary Condition for Energy models.
 
-
-class _BoundaryCondition(object):
-
-    __slots__ = (
-        '_sun_exposure', '_wind_exposure', '_boundary_condition_object', '_view_factor'
-    )
-
-    def __init__(self):
-        self.sun_exposure = False
-        self.wind_exposure = False
-        self.boundary_condition_object = None
-        self.view_factor = 'autocalculate'
-
-    @property
-    def name(self):
-        return self.__class__.__name__
-
-    @property
-    def sun_exposure(self):
-        return self._sun_exposure
-
-    @sun_exposure.setter
-    def sun_exposure(self, value):
-        assert isinstance(value, bool), \
-            'Input value must be a Boolean not {}.'.format(type(value))
-        self._sun_exposure = value
-
-    @property
-    def wind_exposure(self):
-        return self._wind_exposure
-
-    @wind_exposure.setter
-    def wind_exposure(self, value):
-        assert isinstance(value, bool), \
-            'Input value must be a Boolean not {}.'.format(type(value))
-        self._wind_exposure = value
-
-    @property
-    def boundary_condition_object(self):
-        return self._boundary_condition_object
-
-    @boundary_condition_object.setter
-    def boundary_condition_object(self, bco):
-        self._boundary_condition_object = bco
-
-    @property
-    def view_factor(self):
-        return self._view_factor
-
-    @view_factor.setter
-    def view_factor(self, vf):
-        if vf == 'autocalculate':
-            self._view_factor = vf
-        else:
-            self._view_factor = float(vf)
-
-    @property
-    def sun_exposure_idf(self):
-        return 'NoSun' if not self.wind_exposure else 'SunExposed'
-
-    @property
-    def wind_exposure_idf(self):
-        return 'NoWind' if not self.wind_exposure else 'WindExposed'
-
-    @property
-    def boundary_condition_object_idf(self):
-        return '' if not self.boundary_condition_object else \
-            self.boundary_condition_object.name
-
-    @property
-    def to_dict(self):
-        """Boundary condition."""
-        return {
-            'boundary_condition': self.name,
-            'boundary_condition_object': self.boundary_condition_object_idf,
-            'sun_exposure': self.sun_exposure_idf,
-            'wind_exposure': self.wind_exposure_idf,
-            'view_factor': self.view_factor
-        }
-
-    def __repr__(self):
-        return 'BC: %s' % self.name
-
-
-class Surface(_BoundaryCondition):
-
-    __slots__ = ()
-
-    def __init__(self, other_object=None):
-        _BoundaryCondition.__init__(self)
-        self.boundary_condition_object = other_object
+Note to developers:
+    See _extend_honeybee to see how to add these boundary conditions to
+    honeybee.boundarycondition module.
+"""
+from honeybee.boundarycondition import _BoundaryCondition, Surface
 
 
 class Zone(Surface):
```

Comment:
As I said in my earlier comments, I don't think we should support `Zone` boundary conditions now since they are not  compatible with our methods for auto-solving adjacency, calculating closed zone volume, or generating thermal comfort maps.  This is because these `Zone` BCs expect you to specify interior surfaces only once on one of the adjacent zones and then use the `Zone` BC to specify the other adjacent zone.  This means that we will not have have fully closed volume geometries for some zones, creating the issues I note above and generally making it difficult to work with individual zones in isolation of other zones.  We can take it out of this PR now or we could leave it in and I will remove it with my next commit but I just wanted to let you know my thoughts on this.

Reply:
https://github.com/ladybug-tools/honeybee-energy/pull/5/commits/b2c2b596e57c2bea904434f26f4afeac3a6c7c5f